volume/local/local_unix.go

```diff
@@ -112,18 +113,34 @@ func (v *localVolume) mount() error {
 	if v.opts.MountDevice == "" {
 		return fmt.Errorf("missing device in volume options")
 	}
+
 	mountOpts := v.opts.MountOpts
+	mountDevice := v.opts.MountDevice
+
 	switch v.opts.MountType {
-	case "nfs", "cifs":
+	case "nfs":
 		if addrValue := getAddress(v.opts.MountOpts); addrValue != "" && net.ParseIP(addrValue).To4() == nil {
 			ipAddr, err := net.ResolveIPAddr("ip", addrValue)
 			if err != nil {
 				return errors.Wrapf(err, "error resolving passed in network volume address")
 			}
 			mountOpts = strings.Replace(mountOpts, "addr="+addrValue, "addr="+ipAddr.String(), 1)
 		}
+	case "cifs":
+		deviceUrl, err := url.Parse(v.opts.MountDevice)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing mount device url")
+		}
+		if deviceUrl.Host != "" && net.ParseIP(deviceUrl.Hostname()) == nil {
+			ipAddr, err := net.ResolveIPAddr("ip", deviceUrl.Host)
```

Comment:
We also need to strip the port when doing the DNS query:
```suggestion
			ipAddr, err := net.ResolveIPAddr("ip", deviceUrl.Hostname())
```